### PR TITLE
Fix the case where an authenticator may be referenced by JSON pointer, but does not have an ID

### PR DESCRIPTION
**Pull request by mikenachbaur-okta**
Some authenticators may not include an ID at times due to different server policies, but still contain information relevant to objects (remediations or fields) that contain a `relatesTo` property. 

When authenticator information is coalesced, these objects shouldn't be omitted and should still be referenced from their related objects.

### CODE REVIEW

**Review by jaynewstrom-okta — APPROVED** [review]

File: Sources/OktaIdx/Internal/Implementations/Version1/Responses/IDXClient+V1ResponseConstructors.swift [hunk 1]
```diff
@@ -69,28 +69,33 @@ extension V1.Response {
...
+        state = jsonPaths.reduce(into: state, { state, jsonPath in
+            switch jsonPath {
+            case "$.currentAuthenticatorEnrollment":
+                state.append(.enrolling)
+                
+            case "$.currentAuthenticator":
+                state.append(.authenticating)
+                
+            case "$.recoveryAuthenticator":
+                state.append(.recovery)
```
Comment (stevelind-okta) [inline Sources/OktaIdx/Internal/Implementations/Version1/Responses/IDXClient+V1ResponseConstructors.swift:85]: Is it expected that we would have multiple of these? Why not just return early when one is found?

File: Sources/OktaIdx/Internal/Implementations/Version1/Responses/IDXClient+V1ResponseConstructors.swift [hunk 2]
```diff
@@ -69,28 +69,33 @@ extension V1.Response {
...
-            return .authenticating
-        }
-        
-        else if authenticatorEnrollments?.value.first(where: { (authenticator) -> Bool in
-            authenticator.id == authenticatorId
-        }) != nil {
-            return .enrolled
-        }
+    func authenticatorState(for authenticators: [Authenticator],
+                            in jsonPaths: [String]) -> IDXClient.Authenticator.State
```
Comment (stevelind-okta) [inline Sources/OktaIdx/Internal/Implementations/Version1/Responses/IDXClient+V1ResponseConstructors.swift:73]: This function is complex enough that it should probably be unit tested in various permutations?

**Review by stevelind-okta — APPROVED** [review]
Looks good, I think the state checker logic could use some more unit tests.

**Comment by mikenachbaur-okta** [comment]
Agreed; it's actually on my todo list. I'm creating some techdebt tickets for our upcoming sprints for this SDK.

File: Sources/OktaIdx/Internal/Implementations/Version1/Responses/IDXClient+V1ResponseConstructors.swift (hunk 1 above)
Comment (mikenachbaur-okta) [reply Sources/OktaIdx/Internal/Implementations/Version1/Responses/IDXClient+V1ResponseConstructors.swift:85]: Their specificity increases with higher enum values. Ultimately I'd prefer this to be a bitmask, but I need to ensure Objective-C compatibility. This is another tech debt item on my backlog

**Pull request merged**